core/system/src/page_allocator_faults.c

```diff
@@ -48,6 +48,23 @@ uint32_t page_allocator_get_faults(uint8_t page)
     return 0;
 }
 
+
+int page_allocator_check_range_for_box(int box_id, uint32_t start_addr, uint32_t end_addr)
```

Comment:
Add this in a separate commit, but use it from the commit "Implement vmpu_buffer_access_is_ok"

It makes sense to have it a part of the following commit, as we add and use the feature in existing code. A separate commit adds the v7-M implementation of buffer_is_ok, considering the page allocator check from the beginning.
"armv8m: Check page heap in vmpu_buffer_access_is_ok"